starter_code/src/movies.js

```diff
@@ -1,15 +1,373 @@
+    rate: 8.6
+  },
+  {
+    title: "The Silence of the Lambs",
+    year: 1991,
+    director: "Jonathan Demme",
+    duration: "1h 58min",
+    genre: ["Crime", "Drama", "Thriller"],
+    rate: 8.6
+  },
+  {
+    title: 'It"s a Wonderful Life',
+    year: 1946,
+    director: "Frank Capra",
+    duration: "2h 10min",
+    genre: ["Drama", "Family", "Fantasy"],
+    rate: 8.6
+  },
+  {
+    title: "La vita è bella",
+    year: 1997,
+    director: "Roberto Benigni",
+    duration: "1h 56min",
+    genre: ["Comedy", "Drama", "War"],
+    rate: 8.6
+  },
+  {
+    title: "The Usual Suspects",
+    year: 1995,
+    director: "Bryan Singer",
+    duration: "1h 46min",
+    genre: ["Crime", "Drama", "Mystery", "Thriller"],
+    rate: 8.6
+  },
+  {
+    title: "Léon",
+    year: 1988,
+    director: "Luc Besson",
+    duration: "1h 50min",
+    genre: ["Crime", "Drama", "Thriller"],
+    rate: 8.6
+  }
+];
+
+console.log("----------------------------");
+
 // Iteration 1: Ordering by year - Order by year, ascending (in growing order)
+function orderByYear(arr) {
+  if (arr.length == 0) {
+    return [null];
+  } else {
+    arr.sort(function(a, b) {
+      return a.year == b.year ? (a.title > b.title) - 0.5 : a.year - b.year;
+    });
+    return arr;
+  }
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct
 
+function howManyMovies(arr) {
+  let arr2 = [];
+  let arr3 = [];
+  if (arr.length === 0) {
+    return 0;
+  } else {
+    arr2 = arr.filter(a => a.director == "Steven Spielberg");
+    arr3 = arr2.filter(a => a.genre.indexOf("Drama") !== -1);
+    return arr3.length;
+  }
+}
+
 // Iteration 3: Alphabetic Order - Order by title and print the first 20 titles
+function orderAlphabetically(arrNotMutated) {
+  arrTemp = [...arrNotMutated];
+  if (arrTemp.length == 0) {
+    return [null];
+  } else {
+    arrTemp.sort(function(a, b) {
+      return (a.title > b.title) - 0.5;
+    });
+
+    arrTemp.length >= 20 ? (arrTemp = arrTemp.slice(0, 20)) : arrTemp;
+
+    var arrFinal = [];
+
+    arrTemp.forEach(v => arrFinal.push(v.title));
+
+    return arrFinal;
+  }
+}
 
 // Iteration 4: All rates average - Get the average of all rates with 2 decimals
+function ratesAverage(arr) {
+  if (arr.length == 0) {
+    return 0;
+  } else {
+    sum = 0;
+    arr.forEach(v => (!isNaN(v.rate) ? (sum += v.rate) : 1));
+    return Math.round((sum / arr.length) * 100) / 100;
+  }
+}
+
+// console.log (ratesAverage(([{ rate: 8 }, { rate: 8 }])));
 
 // Iteration 5: Drama movies - Get the average of Drama Movies
+function dramaMoviesRate(arr) {
+  if (arr.length == 0 || arr.filter(v => v.genre == "Drama").length == 0) {
+    return 0;
+  } else {
+    var sum = 0;
+    var count = 0;
+    arr
+      .filter(v => v.genre == "Drama")
+      .forEach(v => {
+        if (!isNaN(v.rate)) {
+          sum += v.rate;
+          count++;
+        }
+      });
+    return Math.round((sum / count) * 100) / 100;
+  }
+}
```

Comment:
Just above this function, you already had a function in which you had to just send an array with rates and it would return the average.
So you could first filter the array and then use your ratesAverage() function to get the average of the filtered array.